### ISSUE TITLE
Using TLS options makes redis.connect() not resolve

### ISSUE THREAD

**Issue by nickangelou** [luin/ioredis#683]
Passing in options for `lazyConenct: true` into the Redis constructor means we need to call the `redis.connect()` method manually which is supposed to return a promise. If TLS params are included, the method hangs on the promise resolve.

Example code:
```
const Redis = require('ioredis');
const { readFileSync } = require('fs');

const redis = new Redis({
    host: 'example.com',
    port: 6379,
    password: 'somePass',
    lazyConnect: true,
    // If we cut out `tls` options, redis.connect() returns a promise.
    tls: {
        key: readFileSync('./private.pem'),
        cert: readFileSync('./certificate.crt')
    }
});

const start = async () => {
    try {
        console.log('connecting...');
        // redis.connect().then(() => console.log('really connected')); <--- Never gets there.
        // await redis.connect(); <--- This hangs
        redis.connect(); // <--- This is not a promise anymore, but will allow execution to continue and redis commands will work
        console.log('connected!');
    } catch (err) {
        console.error(err);
    }
};

(start)();
```

**Comment by nickangelou** [comment]
Closing. Need to await on redis.secureConnect()

**Comment by luin** [comment]
Hi, thank you for pointing this out! This issue exists on v3.x and have been fixed on v4.x.

**Comment by luin** [comment]
Just released v4.0.0. Closing this issue.
